### PR TITLE
Implemented the profile details page front end

### LINKED ISSUE #20: FE: Profile Details page
Each Profile should have its own dedicated page that outlines their profile in detail

### PR DESCRIPTION
**Pull request by MilesDev91**
### What this PR does:
- Adds the Profile details page at /profiles/exampleid (will be the profile id after integration)
- Adds the request form. It is complete with validation and is set up to send the proper data to the backend.

### Screenshots / Videos:
image
image
image

### Any information needed to test this feature:
- Navigate to /profile/exampleid to see the page. Resize screen to see responsiveness. You may have to refresh page if resizing with devtools.
- Select/Enter values in the request form to check validation. Start dates should be less than end dates, and all fields are required. Check the console logs on successful submit to check that the proper dates are passed through.

### Any issues with the current functionality:
- The page has a horizontal scrollbar on mobile. The navbar being unresponsive makes it hard to tell if it is my code that is extending the width, but it may need another pass after the navbar is fixed.
- The form has red error messages and the label turns red on invalid input, but I could not get the input border to behave the same. This is probably due to the customization I did on the date picker messing with the normal form behavior.
- The page currently does not have the extra photos below the profile details. This should be easy to implement once it is decided how we are going to implement those photos.

### Review thoughts:
- I hope that using the window.innerWidth is an acceptable way to implement responsiveness in React, it really came in handy.

closes #20

### CODE REVIEW

File: client/src/pages/ProfileDetails/RequestForm/RequestForm.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import { Typography, Button, CircularProgress } from '@mui/material';
+import FormikDatePicker from '../../../components/FormikDatePicker/FormikDatePicker';
+import useStyles from './useStyles';
+
+export default function RequestForm(): JSX.Element {
+  const handleSubmit = (
+    { startDate, endDate }: { startDate: Date; endDate: Date },
+    { setSubmitting }: FormikHelpers<{ startDate: Date; endDate: Date }>,
+  ) => {
+    console.log(startDate, endDate);
```
Comment (sundayezeilo) [inline client/src/pages/ProfileDetails/RequestForm/RequestForm.tsx]: Let's get rid of unnecessary console logs, and ditto for all other places.

File: client/src/pages/ProfileDetails/RequestForm/RequestForm.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    setSubmitting(false);
+  };
+
+  const classes = useStyles();
+
+  const initialDate = () => {
+    console.log(new Date(Date.now()), new Date(Date.now()).getHours());
```
Comment (sundayezeilo) [inline client/src/pages/ProfileDetails/RequestForm/RequestForm.tsx]: Remove this one too

**Review by sundayezeilo — CHANGES_REQUESTED** [review]
Good work with this one :+1:

File: client/src/App.tsx [hunk 3]
```diff
@@ -32,6 +33,7 @@ function App(): JSX.Element {
                 <ProtectedRoute exact path="/dashboard" component={Dashboard} />
                 <ProtectedRoute path="/profile/settings" component={Settings} />
                 <ProtectedRoute path="/bookings" component={Bookings} />
+                <ProtectedRoute path="/profile/exampleid" component={ProfileDetails} />
```
Comment (rajivtitus) [inline client/src/App.tsx]:
```suggestion
                <ProtectedRoute path="/profile/:profile-id" component={ProfileDetails} />
```
We can pass the sitters' profile id as a url param from the listings to the profile details page

File: client/src/components/FormikDatePicker/FormikDatePicker.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  date: Date;
+  setFieldValue: (field: string, value: any) => void;
+  label: string;
+  inputId: string;
+  error?: boolean;
+
+  [inputProps: string]: any;
+}
+
+export default function FormikDatePicker({
```
Comment (rajivtitus) [inline client/src/components/FormikDatePicker/FormikDatePicker.tsx:29]: Good work with this component!

File: client/src/components/FormikDatePicker/FormikDatePicker.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    fontSize: 16,
+    width: '100%',
+    padding: '15px',
+  },
+}));
+
+interface Props {
+  dateField: string;
+  date: Date;
+  setFieldValue: (field: string, value: any) => void;
```
Comment (rajivtitus) [inline client/src/components/FormikDatePicker/FormikDatePicker.tsx]: Won't the `value` be a Date type?

File: client/src/pages/ProfileDetails/ProfileDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+import AvatarDisplay from '../../components/AvatarDisplay/AvatarDisplay';
+import LocationOnIcon from '@mui/icons-material/LocationOn';
+import RequestForm from './RequestForm/RequestForm';
+import useStyles from './useStyles';
+import { useAuth } from '../../context/useAuthContext';
+
+const boxShadow =
+  '0px 0px 1.5px rgba(0, 0, 0, 0.006),0px 0px 3.6px rgba(0, 0, 0, 0.009),0px 0px 6.8px rgba(0, 0, 0, 0.011),0px 0px 12px rgba(0, 0, 0, 0.015),0px 0px 22.1px rgba(0, 0, 0, 0.027),0px 0px 48px rgba(0, 0, 0, 0.1)';
+
+const mockProfile = {
```
Comment (rajivtitus) [inline client/src/pages/ProfileDetails/ProfileDetails.tsx:9]: 👍🏽

File: client/src/pages/ProfileDetails/ProfileDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+  payRate: 14,
+  rating: 4,
+};
+
+export default function ProfileDetails(): JSX.Element {
+  const classes = useStyles();
+
+  return (
+    <PageContainer>
+      <Grid m={window.innerWidth < 600 ? 2 : 4} justifyContent="space-around" container alignItems="flex-start">
```
Comment (rajivtitus) [inline client/src/pages/ProfileDetails/ProfileDetails.tsx]: Using `window.innerWidth` is fine but I'm not sure whether it works when we resize? One option would be to store these values in a state
Ref:  https://www.pluralsight.com/guides/render-window-resize-react

Also, I would recommend looking into the usage of breakpoints and using them where you can.
Ref: https://mui.com/customization/breakpoints/

File: client/src/pages/ProfileDetails/ProfileDetails.tsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+        <Grid
+          xs={12}
+          md={4}
+          item
+          borderRadius={2}
+          boxShadow={window.innerWidth < 600 ? 'none' : 4}
+          container
+          flexDirection="column"
+        >
+          <Typography fontSize="1.2rem" fontWeight="bold" m="3rem auto 1rem auto">
```
Comment (rajivtitus) [inline client/src/pages/ProfileDetails/ProfileDetails.tsx]: I would recommend setting font sizes in variants within MUI's theme and using them across the app to ensure a consistent font size

File: client/src/pages/ProfileDetails/ProfileDetails.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+            ${mockProfile.payRate}/hr
+          </Typography>
+          <Rating sx={{ margin: 'auto' }} value={mockProfile.rating} precision={0.5} />
```
Comment (rajivtitus) [inline client/src/pages/ProfileDetails/ProfileDetails.tsx]:
```suggestion
          <Rating sx={{ margin: 'auto' }} value={mockProfile.rating} precision={0.5} readOnly />
```

File: client/src/pages/ProfileDetails/ProfileDetails.tsx [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+            <LocationOnIcon color="primary" /> &nbsp; {mockProfile.location}
+          </Typography>
+          <Box m={window.innerWidth < 600 ? `${5} ${4}` : 5}>
+            <Typography mb={1} component="h2" fontSize="1rem" fontWeight="bold">
+              About me
+            </Typography>
+            <Typography>{mockProfile.aboutMe}</Typography>
+          </Box>
+        </Grid>
+        {window.innerWidth < 600 ? <hr style={{ width: '100%' }} /> : null}
```
Comment (rajivtitus) [inline client/src/pages/ProfileDetails/ProfileDetails.tsx]: You can use a MUI `Divider` component in place of the HTML `hr` tag

File: client/src/pages/ProfileDetails/RequestForm/RequestForm.tsx [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+            <FormikDatePicker
+              onChange={handleChange}
+              inputId="drop-in"
+              label="Drop In"
+              date={values.startDate}
+              setFieldValue={setFieldValue}
+              dateField="startDate"
+              error={touched.startDate && Boolean(errors.startDate)}
+            />
+            <Typography sx={{ marginBottom: 1, marginTop: -1.5, color: '#d32f2f' }}>{errors.startDate}</Typography>
```
Comment (rajivtitus) [inline client/src/pages/ProfileDetails/RequestForm/RequestForm.tsx]: I would recommend using MUI's theme to apply colors instead

Ditto where applicable

**Comment by rajivtitus** [comment]
@MilesDev91 Good work with this! Please have a look at the comments and address them. Additionally, I would consider refining the UI a bit (additional padding where applicable and restrict the max width of the components to preserve aspect ratio)

File: client/src/components/FormikDatePicker/FormikDatePicker.tsx (hunk 5 above)
Comment (MilesDev91) [reply client/src/components/FormikDatePicker/FormikDatePicker.tsx]: Apparently datepicker passes Date | null, so I changed it to that.

File: client/src/App.tsx (hunk 3 above)
Comment (MilesDev91) [reply client/src/App.tsx]: I believe I changed this in the integration, but I suppose there is no reason to not make this change.

File: client/src/pages/ProfileDetails/ProfileDetails.tsx (hunk 6 above)
Comment (MilesDev91) [reply client/src/pages/ProfileDetails/ProfileDetails.tsx:9]: Ended up removing this in favor of MUI default box shadows

File: client/src/pages/ProfileDetails/ProfileDetails.tsx (hunk 10 above)
Comment (MilesDev91) [reply client/src/pages/ProfileDetails/ProfileDetails.tsx]: Tried this and it wasn't working, I'll try it again.

File: client/src/pages/ProfileDetails/ProfileDetails.tsx (hunk 7 above)
Comment (MilesDev91) [reply client/src/pages/ProfileDetails/ProfileDetails.tsx]: Ah brilliant that is the exact solution I was looking for when I was troubleshooting it. Thanks!

File: client/src/pages/ProfileDetails/ProfileDetails.tsx (hunk 8 above)
Comment (MilesDev91) [reply client/src/pages/ProfileDetails/ProfileDetails.tsx]: Ended up breaking other pages. Should be its own ticket.

**Comment by MilesDev91** [comment]
The requested variant changes ended up breaking other pages. If we have time before the presentation, perhaps we could address this in its own ticket?

**Comment by rajivtitus** [comment]
> The requested variant changes ended up breaking other pages. If we have time before the presentation, perhaps we could address this in its own ticket?

That sounds good to me Miles!

**Review by rajivtitus — APPROVED** [review]
Good work with the changes! 💯 
For subsequent PRs please update the PR template with the updated UI images :)

**Pull request merged**